src/layouts/app/connection-mixin.js

```diff
@@ -144,10 +150,12 @@ export default (superClass) =>
     hassReconnected() {
       super.hassReconnected();
       this._updateHass({ connected: true });
+      fireEvent(document, "connected");
```

Comment:
This file should register a new "connection-status" event. 

```ts
declare global {
  // for fire event
  interface HASSDomEvents {
    "connection-status": string
  }
}
```

Then here call
```suggestion
      fireEvent(document, "connection-status", "connected");
```